### PR TITLE
FLPATH-1856 Enabling monitoring by default in Helm Based Orchestrator operator

### PR DESCRIPTION
**Pull request by jianrongzhang89**
This PR updates the helm based Orchestrator Operator so that the SonataFlowPlatform CR is created with monitoring enabled by default. Users will be able to disable it by setting it to `false`.
```
  orchestrator:
    sonataflowPlatform:
      monitoring:
        enabled: true
```
Tested in RHPDS successfully with the local build of SonataFlow Operator from main branch.

### CODE REVIEW

File: bundle/manifests/orchestrator-operator.clusterserviceversion.yaml
Comment (masayag) [inline bundle/manifests/orchestrator-operator.clusterserviceversion.yaml]: should this be part of this PR?

**Review by masayag — COMMENTED** [review]
Should the PR be rebased? It contains changes that aren't seem to be related to monitoring.

File: bundle/manifests/orchestrator-operator.clusterserviceversion.yaml
Comment (jianrongzhang89) [reply bundle/manifests/orchestrator-operator.clusterserviceversion.yaml]: > should this be part of this PR?

Cleaned up :)

Comment (jordigilh) [reply bundle/manifests/orchestrator-operator.clusterserviceversion.yaml]: Ah I see what happened. The problem is that we no longer generate the bundle as part of the build because Konflux forces us to run hermetic builds and we can't retrieve opm at building time. So we're left with sed/awk to make changes on image and version values. And when we have additional changes like new fields, those are not populated to the bundle files. But if we run the `make bundle`, then these fields now show up in the bundle like something new we changed.

Let me raise a PR now and fix this before yours is merged, then your PR won't have these artifacts.

Comment (jordigilh) [reply bundle/manifests/orchestrator-operator.clusterserviceversion.yaml]: Opened #428 that runs `make bundle` in the existing code. Once it is merged, @jianrongzhang89 please rebase and update the PR, it should have less changes due to running `make bundle` on your side

**Comment by jordigilh** [comment]
> Should the PR be rebased? It contains changes that aren't seem to be related to monitoring.

I've added a comment to explain why these artifacts exists. In a nutshell, yes it should be rebased but only after my PR #147 is merged, which contains the changes that are side effects by Jeff running `make bundle` in his environment and pushing the generated changes in the `bundle/` directory.

**Comment by jenniferubah** [comment]
Also, merged the [CRD v1alpha2 PR](https://github.com/rhdhorchestrator/orchestrator-helm-operator/pull/425) which was blocking this PR. @jianrongzhang89 this should be good to go now whenever it's ready.

File: helm-charts/orchestrator/values.schema.json
Comment (jordigilh) [inline helm-charts/orchestrator/values.schema.json:1206]: You're missing defining these 2 new fields as required, otherwise your conditional check in the `sonataflows.yaml` will fail, well at least the `monitoring` field is mandatory for the conditional to succeed without crashing the templating engine.

Comment (jianrongzhang89) [reply helm-charts/orchestrator/values.schema.json:1206]: Thank you for catching it. Updated :)

File: helm-charts/orchestrator/templates/sonataflows.yaml
Comment (jenniferubah) [inline helm-charts/orchestrator/templates/sonataflows.yaml:27]: Might be missing the addition of this new fields in the chart README file here: https://github.com/rhdhorchestrator/orchestrator-helm-operator/blob/main/helm-charts/orchestrator/README.md

**Review by jenniferubah — APPROVED** [review]
Looks good to me.

**Comment by jenniferubah** [comment]
> /hold until [PR](https://github.com/rhdhorchestrator/orchestrator-helm-operator/pull/377) is merged since it is dependent of having OSL 1.35

No need to hold this PR anymore since we can test it with OSL RC build.

**Pull request merged**
